### PR TITLE
Fix/defense wall

### PR DESCRIPTION
**Pull request by PhiBabin**
- Adapt `defense_wall` to new field dimension
- Remove all the `const["SOMETHING_IN_CAPS"]`
- Add utility function to compute bisection
- Fix bug where the referee commands for the yellow team where interpreted as for the blue team by the autoplay.
- Fix bug in `angle_between_three_points`
- Remove `defense_wall_3v3` and fix `defense_wall_nokick`
- Make the attacker more active in defense wall

### CODE REVIEW

File: ai/GameDomainObjects/referee_state.py [hunk 1]
```diff
@@ -132,7 +132,9 @@ def _parse_command(self, command: RawRefereeCommand):
             else:
                 parsed_cmd = self._convert_raw_to_them(command)
         # None color wise commands
-        return RefereeCommand(parsed_cmd)
+        d =  RefereeCommand(parsed_cmd)
+        print("convert {} to {}".format(command, d))
```
Comment (PhiBabin) [inline ai/GameDomainObjects/referee_state.py]: oups

File: ai/STA/Tactic/goalkeeper.py [hunk 2]
```diff
@@ -128,20 +127,20 @@ def _best_target_into_goal(self):
         ab = a-b
         ac = a-c
 
-        be = self.game_state.field.goal_width / (1 + ab.norm/ac.norm)
+        be = (b-c).norm / (1 + ab.norm/ac.norm)
```
Comment (PhiBabin) [inline ai/STA/Tactic/goalkeeper.py]: Should use the bisection from geometry.

File: ai/GameDomainObjects/field.py [hunk 3]
```diff
@@ -31,12 +31,26 @@ def __init__(self, line: Dict):
         self.thickness = line["thickness"]
 
 
+def convert_field_circular_arc(field_arcs: Dict):
+    result = {}
+    for arc in field_arcs:
+        result[arc["name"]] = FieldCircularArc(arc)
```
Comment (sim0504) [inline ai/GameDomainObjects/field.py]: name represent what?

File: ai/STA/Tactic/align_to_defense_wall.py [hunk 4]
```diff
@@ -19,19 +22,16 @@
 __author__ = 'RoboCupULaval'
 
 ORIENTATION_DEADZONE = 0.2
-DISTANCE_TO_KICK_REAL = ROBOT_RADIUS * 3.4
-DISTANCE_TO_KICK_SIM = ROBOT_RADIUS + BALL_RADIUS
-COMMAND_DELAY = 1.0
 
-DEAD_ZONE = 30  # We lost the true value, but who cares
```
Comment (sim0504) [inline ai/STA/Tactic/align_to_defense_wall.py]: Gab? typo?

File: ai/GameDomainObjects/field.py (hunk 3 above)
Comment (PhiBabin) [reply ai/GameDomainObjects/field.py]: The protobuf arcs list looks like this:
```
[
{
  "name": "CentralCircle",
  "degree": 20,
  "radius": 100,
  ...
},
...
]
````

I want them like this:
```
{"CentralCircle" :
 {
  "degree": 20,
  "radius": 100,
  ...
  },
...
}
```

Comment (sim0504) [reply ai/GameDomainObjects/field.py]: make sense

File: ai/STA/Tactic/align_to_defense_wall.py (hunk 4 above)
Comment (PhiBabin) [reply ai/STA/Tactic/align_to_defense_wall.py]: Gabriel_WALL ? Benoit_WALL?
Mais ouin je vais fix le typo

File: Util/geometry.py [hunk 5]
```diff
@@ -12,15 +12,46 @@ def __init__(self, p1, p2):
...
+
 
 class Area:
     def __init__(self, upper_left, lower_right):
-        self.a = upper_left  # -x, +y
-        self.b = lower_right # +x, -y
+        self.upper_left = upper_left   # -x, +y
+        self.lower_right = lower_right  # +x, -y
 
     def point_inside(self, p: Position) -> bool:
```
Comment (sim0504) [inline Util/geometry.py]: Implementing `__contains__` would have been a cool API for an `Area` :  

```python
if Position(100,300) in Area(Position(0,0), Position(500,500)): ...
```

File: Util/geometry.py [hunk 6]
```diff
@@ -12,15 +12,46 @@ def __init__(self, p1, p2):
         self.p1 = p1
         self.p2 = p2
 
+    @property
```
Comment (sim0504) [inline Util/geometry.py]: an `intersect` method would make a lot a sense here: `Line1.intersect(Line2)`

File: Util/geometry.py [hunk 7]
```diff
@@ -12,15 +12,46 @@ def __init__(self, p1, p2):
...
+    def normalize(self):
+        return normalize(self.p2 - self.p1)
```
Comment (sim0504) [inline Util/geometry.py]: `normalize` seems to ba a strange function to have here, it's more a `direction`. The name direction implicitly say a vector from `p1` to `p2`.

File: Util/geometry.py (hunk 5 above)
Comment (PhiBabin) [reply Util/geometry.py]: Cool shit, I will do it!

File: Util/geometry.py (hunk 7 above)
Comment (PhiBabin) [reply Util/geometry.py]: agreed

**Comment by wonwon0** [comment]
@PhiBabin merge la pull request babin!

**Comment by PhiBabin** [comment]
NO MOM! I WILL MERGE IT WHEN I FEEL LIKE IT!!
Aussi je vais le merge quand j'aurai fait les fixes

**Pull request merged**